### ISSUE TITLE
why return x value, not actual loss in loss functions?

### ISSUE THREAD

**Issue by ChanningPing** [tkazusa/CVAE-GAN#4]
Thank you for sharing this code. I'm just confused about the loss calculation classes. In each call function of them, instead of returning the calculated loss, you returned the input x values. Is there any specific reason for this? Or is there anything I misunderstood about Keras call functions?

```
class ClassifierLossLayer(Layer):
    __name__ = 'classifier_loss_layer'

    def __init__(self, **kwargs):
        self.is_placeholder = True
        super(ClassifierLossLayer, self).__init__(**kwargs)

    def lossfun(self, c_true, c_pred):
        return K.mean(keras.metrics.categorical_crossentropy(c_true, c_pred))

    def call(self, inputs):
        c_true = inputs[0]
        c_pred = inputs[1]
        loss = self.lossfun(c_true, c_pred)
        self.add_loss(loss, inputs=inputs)

        return c_true
```
